src/shared/services/dom-util.service.ts

```diff
@@ -32,6 +32,35 @@ export class DomUtilService {
       if (firstInput) {
         firstInput.focus();
         firstInput.select();
+      } else {
+        firstInput = el.querySelector('div.btn-group') as HTMLInputElement;
+        if (firstInput) {
+          // .open is used for opening the dropdownmenu
+          firstInput.classList.add('open');
+          firstInput = el.querySelector('div.btn-group input') as HTMLInputElement;
+          firstInput.focus();
+          firstInput.select();
+        }
+      }
+    }
+  }
+
+  // TO-DO: find the next editable element in row.
+  focusRightOrLeftCell(index: number, direction: number) {
+    let firstInput = document.querySelector('[tabIndex=\'' + index + '\']') as HTMLElement;
+    if (firstInput) {
+      while (firstInput.nodeName !== 'TD') {
+        firstInput = firstInput.parentElement;
```

Comment:
define a local variable outside of `while` like `parentCell` and assign`.parentElement` to it here. In that case you avoid confusion that is caused by `firstInput` variable being a`td`.